### PR TITLE
bugfix: Float64 error for mps devices on set_timesteps

### PR DESCRIPTION
**Pull request by ZachNagengast**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Community Node Submission

## Have you discussed this change with the InvokeAI team?
- [ ] Yes
- [x] No, because: minor fix, let me know your thoughts

      
## Have you updated all relevant documentation?
- [x] Yes
- [ ] No

## Related Tickets & Documents

- Related Issue # https://github.com/invoke-ai/InvokeAI/issues/4017
- Closes #

## Added/updated tests?

- [ ] Yes
- [x] No : Requires mps device

## [optional] Are there any post deployment tasks we need to perform?

Please test on an MPS (M1/M2) device. 

Relevant code causing the error in #4017 

https://github.com/huggingface/diffusers/blob/01b6ec21faf2dce3373238b12eb450030ab1f318/src/diffusers/schedulers/scheduling_euler_discrete.py#L263C3-L268C75

```
        self.sigmas = torch.from_numpy(sigmas).to(device=device)
        if str(device).startswith("mps"):
            # mps does not support float64
            self.timesteps = torch.from_numpy(timesteps).to(device, dtype=torch.float32)
        else:
            self.timesteps = torch.from_numpy(timesteps).to(device=device)
```

### CODE REVIEW

**Review by Millu — APPROVED** [review]
Just tested this on mine - 2020 M1 w/ 16GB of RAM and it is working as expected.

HUGE shoutout to @psychedelicious for the help in setting up the local dev environment

File: invokeai/app/invocations/sdxl.py [hunk 1]
```diff
@@ -537,27 +538,28 @@ def invoke(self, context: InvocationContext) -> LatentsOutput:
...
             **self.unet.unet.dict(),
             context=context,
         )
+
         do_classifier_free_guidance = True
         cross_attention_kwargs = None
         with unet_info as unet:
+            # apply denoising_start
+            num_inference_steps = self.steps
+            scheduler.set_timesteps(num_inference_steps, device=self.scheduler.device)
```
Comment (wbowling) [inline invokeai/app/invocations/sdxl.py:551]: Should this be `device=unet.device`? `self.scheduler` is a string and `self.scheduler.device` is blowing up for me, although I may have applied the patch incorrectly

Comment (ZachNagengast) [reply invokeai/app/invocations/sdxl.py]: Yes! Missed this one from a previous test, thanks for testing it out 🙌 (fix just pushed)

**Comment by psychedelicious** [comment]
Would like approval from @StAlKeR7779 for this one.

**Review by lstein — CHANGES_REQUESTED** [review]
On a Linux CUDA system, running with this PR gives me this for "banana sushi", SDXL-base-1.0, and 1024x1024 using the Euler scheduler:

image

Same settings on main give me this:
image

**Review by lstein — APPROVED** [review]
With @StAlKeR7779 's help, I committed a small change that fixes the issue on Linux systems. This should be tested again on MPS systems.

**Comment by Millu** [comment]
@ZachNagengast are you in the Discord?

**Comment by zatokar1** [comment]
Thanks a lot @ZachNagengast, I am new to this, but for me this fix doesn't seem to work. I am still getting the error:
"TypeError: Cannot convert a MPS Tensor to float64 dtype as the MPS framework doesn't support float64. Please use float32 instead." Trying to make it work on my Macbook Pro M1.

**Pull request merged**